src/cppsim/simulator.hpp

```diff
@@ -17,6 +17,8 @@ class DllExport QuantumCircuitSimulator {
     QuantumCircuit* _circuit;
     QuantumStateBase* _state;
     QuantumStateBase* _buffer;
+    bool _own_state = false;
+    bool _is_swap = false;
```

Comment:
We should have more direct name like `_is_buffer_used`, shouldn't we?

Reply:
It was. I corrected it because the swap () was wrong, so if the _buffer was reserved, it needed to be opened.
I'll fix it.

Reply:
Now we don't need `_is_swap`.